src/report-filter-controls.js

```diff
@@ -0,0 +1,106 @@
+import { html } from "htm/preact";
+import { createContext } from "preact";
+import { useState } from "preact/hooks";
+import { timeFormat, timeParse } from "d3-time-format";
+import { utcWeek } from "d3-time";
+
+const yearMonthDayFormat = timeFormat("%Y-%m-%d");
+const yearMonthDayParse = timeParse("%Y-%m-%d");
+const DEFAULT_IAL = 1;
+
+const endOfPreviousWeek = utcWeek.floor(new Date());
+const startOfPreviousWeek = utcWeek.floor(new Date(endOfPreviousWeek.valueOf() - 1));
+
+/**
+ * @typedef ReportFilterControlsContextValues
+ * @property {Date} start
+ * @property {Date} finish
+ * @property {1|2} ial
+ * @property {string=} agency
+ * @property {string[]} allAgencies
+ * @property {(s:string[])=>void} setAllAgencies
+ */
+const ReportFilterControlsContext = createContext(
+  /** @type ReportFilterControlsContextValues */ ({
+    start: startOfPreviousWeek,
+    finish: endOfPreviousWeek,
+    ial: 1,
+    allAgencies: [],
+    setAllAgencies: () => null,
+  })
+);
+
+/**
+ * @typedef ReportFilterControlsProps
+ * @property {import('preact').VNode[]} children
+ */
+
+/**
+ * @param {ReportFilterControlsProps} props
+ * @returns {import('preact').VNode}
+ */
+function ReportFilterControls({ children }) {
+  const [allAgencies, setAllAgencies] = useState([]);
+
+  const searchParams = new URLSearchParams(document.location.search);
+
+  const startParam = searchParams.get("start");
+  const finishParam = searchParams.get("finish");
+  const ialParam = searchParams.get("ial");
+
+  const start = (startParam ? yearMonthDayParse(startParam) : null) || startOfPreviousWeek;
+  const finish = (finishParam ? yearMonthDayParse(finishParam) : null) || endOfPreviousWeek;
+  const ial = parseInt(ialParam || "", 10) || DEFAULT_IAL;
+
+  const agency = searchParams.get("agency");
+
+  const filterControls = {
+    start,
+    finish,
+    agency,
+    ial,
+    setAllAgencies,
+  };
+
+  return html`<div>
+    <form>
+      <div>
+        <label>
+          Start
+          <input type="date" name="start" value=${yearMonthDayFormat(start)} />
+        </label>
+      </div>
+      <div>
+        <label>
+          Finish
+          <input type="date" name="finish" value=${yearMonthDayFormat(finish)} />
+        </label>
+      </div>
+      <div>
+        <label><input type="radio" name="ial" value="1" checked=${ial === 1} /> IAL1</label>
+        <label><input type="radio" name="ial" value="2" checked=${ial === 2} /> IAL2</label>
```

Comment:
There's no live updating right now (see note above about URL params), if there's a good way to sync the URL params, I'd totally do controlled, otherwise uncontrolled

Reply:
I think uncontrolled could be fine for this use-case. The concern is if any renders happen, the checked state is going to revert back to whatever `ial` is set to, regardless what changes the user has made in the UI.

e.g. https://codepen.io/aduth/pen/NWgWVJm

Reply:
`defaultChecked` in c6190e13054cde175f9752062d6a7fb6b3931cd8

Reply:
I should have clarified this applies to all fields in this form (e.g. `defaultValue` for start and finish). 

Options are a bit unique, and if it behaves the same as React, would be set as `defaultValue` on the `<select>` tag.

https://reactjs.org/docs/forms.html#the-select-tag

Reply:
the defaultValue for select seems to not work

Reply:
Try `defaultSelected` on `option` ?

https://github.com/preactjs/preact/issues/2646#issuecomment-672135277

Reply:
Winner! 52b1486e749a6ca15080452eb3d394742920882d